src/server/game/World/World.cpp

```diff
@@ -803,6 +803,12 @@ void World::LoadConfigSettings(bool reload)
         TC_LOG_ERROR("server.loading", "InstanceMapLoadAllGrids enabled, but GridUnload also enabled. GridUnload must be disabled to enable instance map pre-loading. Instance map pre-loading disabled");
         m_bool_configs[CONFIG_INSTANCEMAP_LOAD_GRIDS] = false;
     }
+    m_bool_configs[CONFIG_BATTLEGROUNDMAP_LOAD_GRIDS] = sConfigMgr->GetBoolDefault("BattlegroundMapLoadAllGrids", true);
+    if (m_bool_configs[CONFIG_BATTLEGROUNDMAP_LOAD_GRIDS] && m_bool_configs[CONFIG_GRID_UNLOAD])
+    {
+        TC_LOG_ERROR("server.loading", "BattlegroundMapLoadAllGrids enabled, but GridUnload also enabled. GridUnload must be disabled to enable battleground map pre-loading. Battleground map pre-loading disabled");
```

Comment:
We could just ignore GridUnload for battlegrounds when this new config is enabled

Reply:
We can just remove this piece of code here and it will work right? As creatures will be added automatically and keeping the grid active